src/openrct2/network/NetworkBase.cpp

```diff
@@ -3329,26 +3329,36 @@ int32_t network_get_num_players()
 
 const char* network_get_player_name(uint32_t index)
 {
+    Guard::Assert(index >= gNetwork.player_list.size(), "Index %u out of bounds (%zu)", index, gNetwork.player_list.size());
```

Comment:
Does `Guard::Assert` print the calling fn? Or are we expecting to get this from the stack trace?

All of these calls are equal, I'd maybe call this a `checkValidIndex` which would make changing these messages a lot easier in the future.

Reply:
I guess we can add a Guard::CheckBounds(idx, container) or something, the checks are around players and groups.